account_payment_return/models/payment_return.py

```diff
@@ -108,6 +108,13 @@ def _prepare_invoice_returned_cancel_vals(self):
         return {'returned_payment': False}
 
     @api.multi
+    def unlink(self):
+        if self.filtered(lambda x: x.state == 'done'):
+            raise UserError(_(
+                "You can not remove a payment return if state is 'Done'"))
+        return super(PaymentReturn, self).unlink()
```

Comment:
Well, I know it's late and after all it works fine this way, but just to clarify, I meant a *global* rule. Quoting from the above link:

> Global rules are subtractive, they must all be matched for a record to be accessible